File: Online-judge/leetcode/invert-binary-tree.py
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
        
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root == None:
            return root
        left: TreeNode = root.left
        right: TreeNode = root.right
        root.left = self.invertTree(right)
        root.right = self.invertTree(left)
        return root

File: Online-judge/leetcode/invert-binary-tree.cpp
#include <bits/stdc++.h>
using namespace std;

/**
 * @author Md.Shohag <shohag.fks@gmail.com>
 * Problem : https://leetcode.com/problems/invert-binary-tree/description/
 * Complexity: running O(n) 
 * Space O(1)
 */

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (!root) return root;
        TreeNode* left = root->left;
        TreeNode* right = root->right;
        root->left = invertTree(right);
        root->right = invertTree(left);
        return root;
    }
};